src/core/json-schema/src/formly-json-schema.service.ts

```diff
@@ -193,7 +210,27 @@ export class FormlyJsonschema {
     switch (field.type) {
       case 'number':
       case 'integer': {
-        field.parsers = [(v: string | number) => (isEmpty(v) ? v : Number(v))];
+        field.parsers = [
+          (v: string | number) => {
+            v = toNumber(v);
+
+            if (
+              v === null &&
+              typeof document !== 'undefined' &&
+              field.id &&
+              !document.querySelector<HTMLInputElement>(`#${field.id}`)?.validity?.badInput
+            ) {
+              v = undefined;
+            }
+
+            if (v !== field.formControl.value) {
```

Comment:
Since the update i have an issue with this. formControl is not set a this time.
![image](https://github.com/ngx-formly/ngx-formly/assets/64957020/bdf08951-36cc-415a-a241-1fb3c8d79265)

Reply:
@dgilsonAfelio I see. Could you please create a separate ticket ?
if possible please provide a reproduction 🙏